### PR TITLE
atomic_queue: add recipe

### PR DESCRIPTION
**Pull request by luizfeldmann**
### Summary
Specify library name and version: **atomic_queue/1.6.3**

#### Details
C++14 multiple-producer-multiple-consumer lock-free queues based on circular buffer and [std::atomic](https://en.cppreference.com/w/cpp/atomic/atomic). Designed with a goal to minimize the latency between one thread pushing an element into a queue and another thread popping it from the queue.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/atomic_queue/all/test_v1_package/CMakeLists.txt
Comment (uilianries) [inline recipes/atomic_queue/all/test_v1_package/CMakeLists.txt:1]: The entire `test_v1_package` folder can be removed. It's no longer needed for new recipes in CCI.

**Review by uilianries — COMMENTED** [review]
Saudações e obrigado pelo seu PR! 😄 

The `test_v1_package` is used to validate Conan 1.x legacy generators like `cmake_find_package`, but we are dropping that support for new recipes, so you can remove that folder from your branch.

All other things like default CMake target name and requiring C++14 as standard are looking good. We usually impose a list of required compilers x versions to validate it with Conan 1.x in case cppstd is not declared, but GCC 5 passed without errors, so I'm fine skipping it.

Thank you.

**Review by uilianries — APPROVED** [review]
LGTM.

**Review by AbrilRBS — APPROVED** [review]
Thanks! :)

**Pull request merged**